spec/rspec/mocks/partial_double_spec.rb

```diff
@@ -383,6 +383,22 @@ def defined_private_method
         object.implemented
       end
 
+      it 'wraps proxies in a reentrant mutex so we dont trigger deadlocks' do
```

Comment:
We don't have any specs that access `RSpec::Mocks.space.proxy_for` in this file.  This bug kinda straddles things but since we're fixing it by making the space proxy mutex re-entrant it feels to me like it belongs there.

That's subjective, though.  If you feel strongly about it staying here let's go with it.